docs/fsharp-cheatsheet.md

```diff
@@ -169,197 +205,281 @@ The `rec` keyword is used together with the `let` keyword to define a recursive
         if x = 0 then false
         else even (x - 1)
 
-<div id="pattern-matching"></div>
-
-## Pattern Matching
+<div id="statically-resolved-type-parameters"></div>
 
-Pattern matching is often facilitated through `match` keyword.
+### Statically Resolved Type Parameters
 
-    let rec fib n =
-        match n with
-        | 0 -> 0
-        | 1 -> 1
-        | _ -> fib (n - 1) + fib (n - 2)
+A *statically resolved type parameter* is a type parameter that is replaced with an actual type at compile time instead of at run time. They are primarily useful in conjunction with member constraints.
 
-In order to match sophisticated inputs, one can use `when` to create filters or guards on patterns:
+    let inline add x y = x + y
+    let integerAdd = add 1 2
+    let floatAdd = add 1.0f 2.0f // without `inline` on `add` function, this would cause a type error
 
-    let sign x =
-        match x with
-        | 0 -> 0
-        | x when x < 0 -> -1
-        | x -> 1
+####
 
-Pattern matching can be done directly on arguments:
+    type RequestA = { Id: string; StringValue: string }
```

Comment:
your anon type above has camelCase
Should either stick with FDG rules and be PascalCase for all records, or go camelCase for all in this guide (I think records should be camelCase myself)

Reply:
Likely a typo; will review.

Reply:
I think they should be camel as well, but I'll stick with what F# docs goes with. Thanks for the catch.